plext/using_plpython.html.md.erb

```diff
@@ -2,374 +2,608 @@
 
-The HAWQ PL/Python language module imports the Python module `plpy`. The module `plpy` implements these functions:
+1. Create a database named `testdb`:
 
-- Functions to execute SQL queries and prepare execution plans for queries.
-   - `plpy.execute`
-   - `plpy.prepare`
-   
-- Functions to manage errors and messages.
-   - `plpy.debug`
-   - `plpy.log`
-   - `plpy.info`
-   - `plpy.notice`
-   - `plpy.warning`
-   - `plpy.error`
-   - `plpy.fatal`
-   - `plpy.debug`
+    ``` shell
+    gpadmin@hawq-node$ createdb testdb
+    ```
+
+1. Create a table named `sales`:
+
+    ``` shell
+    gpadmin@hawq-node$ psql -d testdb
+    ```
+    ``` sql
+    testdb=> CREATE TABLE sales (id int, year int, qtr int, day int, region text)
+               DISTRIBUTED BY (id);
+    ```
+
+2. Insert data into the table:
+
+    ``` sql
+    testdb=> INSERT INTO sales VALUES
+     (1, 2014, 1,1, 'usa'),
+     (2, 2002, 2,2, 'europe'),
+     (3, 2014, 3,3, 'asia'),
+     (4, 2014, 4,4, 'usa'),
+     (5, 2014, 1,5, 'europe'),
+     (6, 2014, 2,6, 'asia'),
+     (7, 2002, 3,7, 'usa') ;
+    ```
+
+## <a id="pymod_intro"></a>Python Modules 
+A Python module is a text file containing Python statements and definitions. Python modules are named, with the file name for a module following the `<python-module-name>.py` naming convention.
+
+Should you need to build a Python module, ensure that the appropriate software is installed on the build system. Also be sure that you are building for the correct deployment architecture, i.e. 64-bit.
+
+### <a id="pymod_intro_hawq"></a>HAWQ Considerations 
+
+When installing a Python module in HAWQ, you must add the module to all segment nodes in the cluster. You must also add all Python modules to any new segment hosts when you expand your HAWQ cluster.
+
+PL/Python supports the built-in HAWQ Python module named `plpy`.  You can also install 3rd party Python modules.
+
+
+## <a id="modules_plpy"></a>plpy Module 
+
+The HAWQ PL/Python procedural language extension automatically imports the Python module `plpy`. `plpy` implements functions to execute SQL queries and prepare execution plans for queries.  The `plpy` module also includes functions to manage errors and messages.
    
-## <a id="executepreparesql"></a>Executing and Preparing SQL Queries 
+### <a id="executepreparesql"></a>Executing and Preparing SQL Queries 
 
-The PL/Python `plpy` module provides two Python functions to execute an SQL query and prepare an execution plan for a query, `plpy.execute` and `plpy.prepare`. Preparing the execution plan for a query is useful if you run the query from multiple Python functions.
+Use the PL/Python `plpy` module `plpy.execute()` function to execute an SQL query. Use the `plpy.prepare()` function to prepare an execution plan for a query. Preparing the execution plan for a query is useful if you plan to run the query from multiple Python functions.
 
```

Comment:
Change "an SQL" -> "a SQL"  I'm pretty sure the HAWQ docs are otherwise consistent in this usage.  

Also, change "if you plan to" to "if you want to"